changes made to build without warnings

aditya-vk opened the pull request:
typedef (unused) -> set to ignore this warning. Need a fix.

Major warnings: 
different types compared -> type casted as appropriate
incorrectly ordered initialization -> reordered
unused variables -> specified to have been unused

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/personalrobotics/aikido/143)
<!-- Reviewable:end -->

mkoval commented on src/constraint/uniform/RnBoxConstraint.cpp:
```diff
@@ -62,7 +62,7 @@ bool RnBoxConstraintSampleGenerator::sample(
   Eigen::VectorXd value(mDistributions.size());
 
-  for (size_t i = 0; i < value.size(); ++i)
+  for (int i = 0; i < value.size(); ++i) // i should be the same type as size()
```
I don't think this comment is necessary.

aditya-vk commented on src/trajectory/Interpolated.cpp:
```diff
@@ -90,11 +90,11 @@ void Interpolated::evaluate(double _t, State *_state) const
   Eigen::VectorXd& _tangentVector ) const
 {
-  if (_derivative == 0)
+  if (_derivative <= 0) // Invalid
```
I didn't realize I pushed this edit as well. This is not an edit that accounts for warnings. Do you recommend I remove it?

aditya-vk replied:
I didn't realize I pushed this edit as well. This has nothing to do with warnings. Recommend I remove it?

aditya-vk commented:
1. static_cast of variables - check
2. ignore external project for warnings - check

mkoval commented on CMakeLists.txt:
```diff
@@ -137,7 +137,7 @@ include_directories(${DART_INCLUDE_DIRS})
 #
 # TODO: Add this to each target's DEFINITIONS property.
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Werror -Wno-unused-local-typedefs")
```
Why is `-Wno-unused-local-typedefs` necessary? Could we remove the unused `typedef`s instead? 😉

mkoval commented on src/statespace/dart/SE3Joint.cpp:
```diff
@@ -6,8 +6,8 @@ namespace dart {
-  : JointStateSpace(_joint)
-  , SE3()
+  : SE3()
+  , JointStateSpace(_joint) // Order of class members to be maintained
```
This comment is misleading. `SE3` and `JointSubSpace` are super classes, not class members.

aditya-vk replied:
Ah. My bad. I'll fix it.

jslee02 commented:
Can we merge this after #114 is merged? I'd like to make `-Werror` optional (default is off) and turn it on for CI tests.

mkoval commented:
I approved #114. Let's merge that first, make the last few changes here, and merge this as well.

aditya-vk commented:
I'll review this again first thing in the morning, address the suggestions by @mkoval. Should be ready for merge by morning.

aditya-vk commented:
Should be ready for merge now. :sweat_smile: 
- Removed unnecessary (and um erroneous) comments which weren't required
- Removed the unused typedef and corresponding flag for compiling

jslee02 commented:
Let me resolve the conflicts.

jslee02 commented on src/statespace/dart/SE2Joint.cpp:
```diff
@@ -6,8 +6,9 @@ namespace dart {
-  , SE2()
+  : SE2()
+  , JointStateSpace(_joint)
+  
```
Nit: Remove unnecessary empty line

aditya-vk commented:
Builds without warnings even on unit tests. 
ToDo: Run the same with -Wpedantic for a stricter check.

aditya-vk commented:
I am not sure what to do with the following warning:
`error: anonymous variadic macros were introduced in C99 [-Werror=variadic-macros]`

Rest of the warnings have all been fixed with -Wpedantic as well.

mkoval commented on src/external/hauser_parabolic_smoother/CMakeLists.txt:
```diff
@@ -11,4 +11,5 @@ set_target_properties("${PROJECT_NAME}_external_hauserparabolicsmoother"
 )
 target_compile_options("${PROJECT_NAME}_external_hauserparabolicsmoother"
   PUBLIC ${AIKIDO_CXX_STANDARD_FLAGS}
+  PRIVATE -w
```
What is this? 😱

jslee02 replied:
If I'm correct, this is to ignore warnings from `${PROJECT_NAME}_external_hauserparabolicsmoother` target. Since it's not an external target, we cannot use `include_directories()` with `SYSTEM` option.

mkoval commented on src/trajectory/Interpolated.cpp:
```diff
@@ -90,11 +90,11 @@ void Interpolated::evaluate(double _t, State *_state) const
   Eigen::VectorXd& _tangentVector ) const
 {
-  if (_derivative == 0)
+  if (_derivative == 0) // Invalid
```
Nit: Remove comment.

mkoval commented on tests/constraint/test_SO2UniformSampler.cpp:
```diff
@@ -26,7 +26,7 @@ class SO2UniformSamplerTests : public ::testing::Test
 
     mTargets.clear();
-    for (int i = 0; i < NUM_TARGETS; ++i)
+    for (auto i = 0u; i < NUM_TARGETS; ++i)
```
Nit: Could we just make this a `size_t`? I'm generally a fan of [almost always `auto`](https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/), but it's a bit funny that we changed a bunch of other loop variables to `size_` in this pull request. 😉

aditya-vk commented:
Woohoo! Done! I guess? :D

jslee02 commented on tests/CMakeLists.txt:
```diff
@@ -1,5 +1,6 @@
 # GTest setup.
 add_subdirectory(gtest)
+
 include_directories("${gtest_SOURCE_DIR}/include")
```
Nit: This line is not added by this PR, but it seems redundant. `"${gtest_SOURCE_DIR}/include"` is included in the `CMakeLists.txt` under `gtest` subdirectory.

jslee02 commented on tests/constraint/test_TSR.cpp:
```diff
@@ -18,15 +18,15 @@ using dart::common::make_unique;
 
 
-static Eigen::Vector6d getPose(Eigen::Isometry3d isometry)
+static inline Eigen::Vector6d getPose(Eigen::Isometry3d isometry)
```
Does this really cause warnings? This function is defined in source code so we shouldn't worry about duplicated definitions. Am I missing something here?

aditya-vk replied:
From the documentation:
**-Wunused-function**
Warn whenever a static function is declared but not defined or a non-inline static function is unused. This warning is enabled by -Wall. 
In our case a non-inline static function's not used.

jslee02 replied:
It seems the real problem is this function is _unused_. I think we could simply remove this function since it was defined only for the local tests but not used.

jslee02 commented on cmake/CompilerSettings.cmake:
```diff
@@ -17,6 +17,8 @@ if(CMAKE_COMPILER_IS_GNUCXX)
   set(AIKIDO_CXX_STANDARD_FLAGS -std=c++11)
 
+  add_compile_options(-Wall)
+  add_compile_options(-Wpedantic)
```
Nit: We could put multiple options in one function like ```add_compile_options(-Wall -Wpedantic)```.

aditya-vk replied:
I don't see any errors, that's odd. Shall meet you to sort them out.

jslee02 commented on tests/constraint/test_NonColliding.cpp:
```diff
@@ -35,17 +35,11 @@ class NonCollidingTest : public ::testing::Test
     Eigen::Vector3d boxSize(0.5, 0.5, 0.5);
     std::shared_ptr<BoxShape> boxShape(new BoxShape(boxSize));
-    auto shapeNode = boxNode->createShapeNodeWith
-                      <VisualAspect, CollisionAspect, DynamicsAspect>(boxShape);
```
You shouldn't remove the whole line, but just `shapeNode` variable. `createShapeNodeWith<...>()` creates `ShapeNode` in `boxNode` and returns the pointer. That means this function changes the internal state of `bodyNode`, which is necessary for this test.

jslee02 commented on tests/constraint/test_NonColliding.cpp:
```diff
@@ -35,17 +35,11 @@ class NonCollidingTest : public ::testing::Test
-    // Create a shpae node for visualization and collision checking
-    auto shapeNode1 = bn1->createShapeNodeWith
-                      <VisualAspect, CollisionAspect, DynamicsAspect>(box);
-
```
Please don't remove the whole line but `shapeNode1`. (see above)

jslee02 commented:
All the warnings of more tighter warning levels are gone! :tada:

jslee02 commented:
@mkoval Could you do take a look once more for sure?
